lib/recorder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <memory>
+
+namespace swss {
+
+class RecBase {
+public:
+    RecBase() {}
+    /* Setters */
+    void setRecord(bool record)  { enable_rec = record; }
+    void setRotate(bool rotate)  { log_rotate = rotate; }
+    void setLocation(const std::string& loc) { location = loc; }
+    void setFileName(const std::string& name) { filename = name; }
+    void setName(const std::string& name)  { m_name = name; }
+
+    /* getters */
+    bool isRecord()  { return enable_rec; }
+    bool isRotate()  { return log_rotate; }
+    std::string getLoc() { return location; }
+    std::string getFile() { return filename; }
+    std::string getName() { return m_name; }
+
+private:
+    bool enable_rec;
+    bool log_rotate;
+    std::string location;
+    std::string filename;
+    std::string m_name;
+};
+
+class RecWriter : public RecBase {
+public:
+    RecWriter() : RecBase() {}
+    virtual ~RecWriter();
+    void startRec(bool exit_if_failure);
+    void record(const std::string& val);
+
+protected:
+    void logfileReopen();
+
+private:
+    std::ofstream record_ofs;
+    std::string fname;
+};
+
+class SwSSRec : public RecWriter {
+public:
+    SwSSRec();
+};
+
+/* Record Handler for Response Publisher Class */
+class ResPubRec : public RecWriter {
+public:
+    ResPubRec();
+};
+
+class SaiRedisRec : public RecBase {
+public:
+    SaiRedisRec();
+};
+
+/* Interface to access recorder classes */
+class Recorder {
+public:
+    static const std::string DEFAULT_DIR;
+    static const std::string REC_START;
+    static const std::string SWSS_FNAME;
+    static const std::string SAIREDIS_FNAME;
+    static const std::string RESPPUB_FNAME;
+    
+    static std::unique_ptr<SwSSRec> swss;
+    static std::unique_ptr<SaiRedisRec> sairedis;
+    static std::unique_ptr<ResPubRec> respub;
```

Comment:
i would prefer to not have those static objects

Reply:
What would you recommend? The only other option i could think of is to define a singleton of Recorder and have these as public member objects.

But to access the singleton everytime, we might need to define a method like getInstance() and use it to access the object and their methods, kinds like the logger class under swsscomon

Eg:` Recorder::getInstance().swss.startRec()`

Reply:
you dont need actually singleton, just a instance of recorder in Orch class, if you have more static members, its harder to test

Reply:
Defining recorder instance as a member of Orch class will not work. There are multiple orch classes and all write to a same file. 

So, logically the scope of recorder is the daemon/process and not just orch. Even if we can workaround with using an instance of the recorder class per orch, we would still need static members for handing ofstream, path etc. 

Best case scenario is one static Recorder instance and using it as singleton.

Reply:
ok